### ISSUE TITLE
undervolt _ No changes for me !!! Can i get some help ?

### ISSUE THREAD

**Issue by dlj2019** [xdever/linux-intel-undervolt-tool#2]
Can someone please let know if i am adding the correct values to .json files ?

{
  "cpu": -100,
  "gpu": -100
}

Thanks in advance !

**Comment by RobertCsordas** [comment]
How are you checking whether something changed?

**Comment by dlj2019** [comment]
Yes... i trying to see if is there something else i am missing here...

{
"cpu": -100mV,
"gpu": -100mV
}

Should i also add "mV"  with the value ?

Thanks gain

**Comment by dlj2019** [comment]
> How are you checking whether something changed?

Yes... i trying to see if is there something else i am missing here...

{
"cpu": -100mV,
"gpu": -100mV
}

Should i also add "mV" with the value ?

Thanks gain

**Comment by ariselseng** [comment]
@dlj2019 You need valid JSON. Remove the "mV".

**Comment by RobertCsordas** [comment]
The JSON in the first post is correct:

{
"cpu": -100,
"gpu": -100
}

**Comment by dlj2019** [comment]
I had follow the steps but the report shows;


![Selection_038](https://user-images.githubusercontent.com/30886694/55670431-2900ed80-5852-11e9-8ad2-ca47feebccc3.png)

Can you guys help me out !?

Thanks in advance

> The JSON in the first post is correct:
> 
> {
> "cpu": -100,
> "gpu": -100
> }

I had follow the steps but the report shows;


![Selection_038](https://user-images.githubusercontent.com/30886694/55670431-2900ed80-5852-11e9-8ad2-ca47feebccc3.png)

Can you guys help me out !?

Thanks in advance

**Comment by dlj2019** [comment]
> The JSON in the first post is correct:
> 
> {
> "cpu": -100,
> "gpu": -100
> }

Unfortunately, i still see high discharge rate on XPS 15 9560- KDE Neon.

![Selection_038](https://user-images.githubusercontent.com/30886694/55671781-86e90180-5861-11e9-99ff-904cbfbcc915.png)

Can you please help me out!

I really appreciate you time...

Thanks

**Comment by RobertCsordas** [comment]
It is highly task dependent whether you will see any decrease in battery discharge rate. You have more serious problems here. 45.7W is incredibly high. I usually get 5-6W on iddle with minimum brightness. I have never measured the reduction in power consumption due to this script, however I noticed that under load the machine runs a bit cooler. If you set ~ -150 for the cpu, you will notice that your system hangs if you do some really CPU intensive, so I's definitely reducing the CPU voltage. There is an other, more spohisticated project (https://github.com/kitsunyan/intel-undervolt), which is about a month newer than this one, and can also read the limits. However they just re-read the same registers as there are written by this script. However note that the other one lets you to separately control the CPU and cache voltages, which is by current hardware handled by max(cpu, cache), e.g for -100 and -50, it will just set -50 for both.

You should install TLP which helps a lot in power consumtion. However your nic:virbr0 should should not use power when it is not doing anything, and here it uses 16.5W. Do you really need that virtual ethernet device? Also what is that usb device using 3.5W?

Here is a screenshot of mine:
![power](https://user-images.githubusercontent.com/2706617/55680554-58c7f800-591b-11e9-801a-27ebc14d170c.png)

**Comment by dlj2019** [comment]
Wow... 5-6W ? that's awesome ! i wish i could get it close to that!

Powertop is installed and running ...

Right, i am not using the "nic:virbr0"... i need to disable it. For the "usb" , not so sure what it is.
Battery is reporting much better readings now compering to 45.7w, but there is much more room for improvement .

**Comment by RobertCsordas** [comment]
Make sure you turn off your discrete GPU. You can configure bumblebee to switch between them. I just use it for CUDA, and I wrote a script that can turn it on/off easily. If you are interested, I can share that also. I think if you remove the virbr, install TLP and turn off your GPU, it should be the same. One more thing: I have my touchscreen disabled from BIOS, which helps a tiny bit (<0.5W), although it you add it to TLP config then it also uses almost nothing. Take a look at this page for instructions (it should work for other distros too): https://wiki.archlinux.org/index.php/Dell_XPS_15_9560

**Comment by dlj2019** [comment]
Thank you so much for helping me out with this. I really appreciate your help.

I want to take you advice, could you please share the script you made ?
I think i already have TLP installed. ( i share a screenshot of it )
Thanks again for your time ...

**Comment by dlj2019** [comment]
Question:

What do you have as Grub set (/etc/default/grub)?
GRUB_CMDLINE_LINUX_DEFAULT="quiet splash nouveau.modeset=0 acpi_rev_override=5 acpi_backlight=none"

**Comment by RobertCsordas** [comment]
For the switching tool, see https://github.com/xdever/linux_gpu_switch
My kernel flags are: "acpi_rev_override=5 i915.enable_fbc=1 i915.enable_psr=1  iwlwifi.power_save=Y iwlwifi.power_level=5 i915.fastboot=1 i915.enable_guc=2".
By the way I doesn't use GRUB, but boot directly from EFI: https://github.com/xdever/arch-efiboot. This makes no difference in power usage, it is just faster and cleaner in my opinion.
